pkg/githubreconciler/issuemanager/session.go

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright 2025 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package issuemanager
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/chainguard-dev/clog"
+	"github.com/chainguard-dev/terraform-infra-common/pkg/githubreconciler"
+	"github.com/google/go-github/v75/github"
+)
+
+// IssueSession represents work on multiple issues for a specific resource path.
+// Unlike Session in changemanager which handles a single PR, IssueSession manages multiple issues.
+// T must implement the Comparable interface to enable matching between existing and desired issues.
+type IssueSession[T Comparable[T]] struct {
+	manager        *IM[T]
+	client         *github.Client
+	resource       *githubreconciler.Resource
+	pathLabel      string
+	existingIssues []*github.Issue
+}
+
+// hasSkipLabel checks if a specific issue has the skip label.
+// Returns true if the issue has a label matching "skip:{identity}".
+func (s *IssueSession[T]) hasSkipLabel(issue *github.Issue) bool {
+	skipLabel := "skip:" + s.manager.identity
+	for _, label := range issue.Labels {
+		if label.GetName() == skipLabel {
+			return true
+		}
+	}
+	return false
+}
+
+// UpsertMany creates or updates multiple issues based on the provided data.
```

Comment:
I think that we actually want to combine Upsert/Close for this reconciler.

Basically what we are saying is "These are the N desired issues" which should result in:
1. Leaving any existing issues with the skip label alone (no closing/updating),
2. Creating any issues that don't exist,
3. Updating any issues that do exist (but aren't the right shape),
4. Deleting any issues that exist, but aren't in our desired set.